### ISSUE TITLE
Handle Plotly animation

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#1773]
### Description

The goal would be to handle [Plotly animation](https://plotly.com/python/animations/) inside Taipy or find a workaround for it.

This was asked by a user.

### Technical proposal 1 [DOES NOT WORK]

We seem to cover basic use cases with the addition of an `animation` property that would describe if and how animation would take place.
This property must indicate which dataset columns are animated, and animation properties such as the transition easing and speed.

Use case: I want to change which of the dataset column is represented, animating from the initial one to the next.
Data definition is:
```python
     x = [0, 1, 2, 3]
     y1 = [0, 1, 4, 9]
     y2 = [0, 1, 8, 27]
```
Animated column value: `y_values="y1"`

Chart definition is:
```
      x=x 
      y={y_values}  (so it can be dynamically updated)
      animation={ on="y", duration=1000, easing="linear" }
```
Meaning:
  We are initially showing 'y1' on the 'y' axis.
  If 'y' changes to a new column, animate the chart with the indicated parameters.
Now I set 'y_values' to "y2", the chart can animate from the values in 'y1' to the values in 'y2'.

#### Why it does not work

The problem is `y={y_values}` that presumes that `y` is dynamic, which it is not.
Although this would be the panacea, it is not how Taipy GUI charts work, and it would be a mess to change that.

### Technical proposal 2

Data definition is:
```python
data = {
     "x": [0, 1, 2, 3],
     "y": [0, 1, 4, 9]
}
```

We will add an `animation_data` dynamic property that must be a dataset with only the columns that are animated:
```python
data = {
     "y": [0, 1, 8, 27]
}
```

When the chart component has this property set, it loads it all, and receives all the columns that are impacted.
Because the "y" columns appears in both datasets, the 'from' and 'to' frames can be built from the two column values.
And animation is performed.

When a new `animation_data` is sent (with new data, in the same column(s)), we move the 'to' frame to the 'from' frame and recompute the new 'to' frame from the values in the `animation_data` column(s).

From the user stand point, all that needs to be done is update the `animation_data` content. The animation is carried immediately.
There should also be a `animation_config` (or `animation` for short) dictionary (static) property that can tune the animation parameters such as speed or type.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by ScalarJerk** [comment]
@FlorianJacta to clarify, we're adding a gui element that allows users to integrate plotly animations, right?

**Comment by FlorianJacta** [comment]
@ScalarJerk I think it could be taken in charge by the chart visual element. What do you think? @FabienLelaquais @FredLL-Avaiga

**Comment by rakshitmalik136** [comment]
Hey! @ScalarJerk and @FlorianJacta 
I want to work on this issue under hacktoberfest. Please assign me this..

**Comment by FlorianJacta** [comment]
@ScalarJerk Was your plan to work on this issue?

**Comment by rakshitmalik136** [comment]
> Hey! @ScalarJerk and @FlorianJacta I want to work on this issue under hacktoberfest. Please assign me this..

@ScalarJerk @FlorianJacta any updates??
shall i start working on it?

**Comment by FlorianJacta** [comment]
Yes, I think you can start working on it. I am assigning you this issue @rakshitmalik136 ! Thank you

**Comment by rakshitmalik136** [comment]
Just to Clarify, the code that i need to work on, should reside in taipy\frontend\taipy-gui\public\stylekit\elements folder or somewhere else?
@FlorianJacta 


taipy\frontend\taipy-gui\.../chart.tsx more

**Comment by THEBOSS0369** [comment]
Hey @rakshitmalik136 ! Are you able to start the server i.e, the gui server for the front-end, if yes can you help me i am unable to start the server. I have been stuck in this problem for 4 days . Please contact me on the discord
[This issue on discord](https://discord.com/channels/1125797687476887563/1179756453771358228/threads/1291254801945985065)

**Comment by FredLL-Avaiga** [comment]
> Hey @rakshitmalik136 ! Are you able to start the server i.e, the gui server for the front-end, if yes can you help me i am unable to start the server. I have been stuck in this problem for 4 days . Please contact me on the discord
> [This issue on discord](https://discord.com/channels/1125797687476887563/1179756453771358228/threads/1291254801945985065)

You can look at simple example in doc/gui/examples

**Comment by FredLL-Avaiga** [comment]
> @FlorianJacta to clarify, we're adding a gui element that allows users to integrate plotly animations, right?

The chart component is located at  taipy/frontend/taipy-gui/.../chart.tsx

**Comment by rakshitmalik136** [comment]
@quest-bot loot #1773

**Comment by mzf11125** [comment]
Hey, I'm interested in helping with this issue. Is it possible? Thanks

**Comment by mzf11125** [comment]
@quest-bot embark
